vulcan/lib/client/jsx/components/workflow/user_interactions/inputs/input_types.ts

```diff
@@ -15,6 +15,30 @@ export type InputType = string;
 // This type represents the unflattened raw data format coming in from the cwl.
 export type DataEnvelope<Inner> = { [k: string]: Inner };
 
+export function get_vals(de: DataEnvelope<any>): any[] {
+  // could add any[] typing to the return if that's not automatic?
+  let vals = new Array();
+  for (var key in de) {
+    vals.push(de[key]);
+  }
+  return vals
+}
+
+export function get_keys(de: DataEnvelope<any>): string[] {
+  // could add string typing to the return if that's not automatic?
+  let keys = new Array();
+  for (var key in de) {
+    keys.push(key);
+  }
+  return keys
+}
+
+export function nulled_vals(de: DataEnvelope<any>): DataEnvelope<null> {
+  let de_ = de;
```

Comment:
As written, `de` will still be mutated, since you are still preserving the original reference.  i worry that will introduce mutation state bugs.  Let's change this line to `let de_ = {...de};` -- this will make a shallow copy (which is fine for referential integrity, we're not mutating any inner values themselves.